Workflow to deploy website

gmealy1 opened the pull request:
# Description of the issue
Website does not build and deploy automatically on the branch.

# Description of changes
Adds workflow file that builds and deploys website on commit to branch.

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
_Describe what tests you have done._

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make linter`

SaxyPandaBear commented on .github/workflows/deploy-to-gh-pages.yml:
```diff
@@ -0,0 +1,29 @@
+
+
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@v4
```
Is there not a GitHub approved action for this? I'm a little concerned about adding this.

gmealy1 replied:
Github even uses this action as an example here. Not sure if that counts as official.
https://github.com/marketplace/actions/deploy-to-github-pages

gmealy1 replied:
Looking for alternatives now.

gmealy1 commented:
There were build warnings (which make the build fail to compile) so I fixed them just now too, then my formatter fixed it up too, sorry for the extra changes. Just removed like 3 variables and added another = to a comparison.

SaxyPandaBear reviewed (approved):
LGTM but I did skipped over the linting/formatting
